master/buildbot/db/logs.py

```diff
@@ -345,6 +345,40 @@ def thdcompressLog(conn):
         saved = yield self.db.pool.do(thdcompressLog)
         defer.returnValue(saved)
 
+    def deleteOldLogChunks(self, older_than_timestamp):
+        def thddeleteOldLogs(conn):
+            model = self.db.model
+            res = conn.execute(sa.select([sa.func.count(model.logchunks.c.content)]))
+            count1 = res.fetchone()[0]
+            res.close()
+
+            # update log types older than timestamps
+            res = conn.execute(
+                model.logs.update()
+                .where(model.logs.c.stepid.in_(
+                    sa.select([model.steps.c.id])
+                    .where(model.steps.c.started_at < older_than_timestamp)))
+                .values(type='d')
+            )
+            res.close()
+
+            # query all logs older than timestamps
+            q = sa.select([model.logs.c.id])
```

Comment:
I think this should not change any thing unless we add an index to either column (and we probably should)